zh/tidb-database-development-specification/mitigation-of-hot-issues.md

```diff
@@ -0,0 +1,81 @@
+# 十三、缓解热点问题
+
+## 1. 写入热点
+
+大部分单机 RDBMS 采用 B+ tree 数据结构，主键往往是用于组织数据的关键索引（此时表被称作索引组织表），同一数据页内的记录按主键顺序存放。因此单机 RDBMS 产品一般推荐写入连续的序列号，这样每次写入新的记录，都会顺序添加到当前 B+ tree 索引节点的后续位置，当前的数据页写满时，会自动开始新一页的写入。相反，过于随机的主键值，会导致新记录被写入到数据页的某个中间位置，造成数据的移动而带来了额外的开销。
+
+虽然 TiDB 具有不同于单机 RDBMS 的数据结构，但顺序的主键值写入，在 TiDB 上也会产生类似的效果：TiKV 上一个的 region 被写满，进而分裂出一个新的 region，后续的写入转由新的 reigon 来承载。但甲之蜜糖，乙之砒霜，单机 RDBMS 的最佳实践放到 TiDB 上，会使写入压力总是集中在一个 region 上，这样就构成了持续的写入热点，无法发挥出 TiDB 这种分布式数据库的并行写入能力，降低了业务写入瓶颈，造成了系统资源的浪费。
+
+TiDB 发生写入热点的原因主要有以下几种：
+
+- 小表，整个表只有一个 region，高并发的请求会造成读写热点。
+
+- 大量写入时 Key 值的离散程度不足以跳过一个 region 大小（96MB）而引起写入热点。
+
+## 2. 热点的观测和定位
+
+从 TiDB v4.0 版本开始提供便于迅速识别集群负载的 Dashboard 流量可视化页面（Key Visualizer），下图展示了写入热点的显示效果，中间一条明亮的曲线即标志着存在一张连续写入 Key 值的表。而右上侧的一组线条则显示出一个写入压力较为均匀的负载。Key Visualizer 的具体使用方法请参考[官方文档](https://docs.pingcap.com/zh/tidb/stable/dashboard-key-visualizer)。
+
+<img src="https://lh6.googleusercontent.com/nTz918HQkl7pwgbUyge35grdq0LfONrwDBZCC1t8hhVWodhv2VRlxTCAJYK-AjRbH_MIikrAG8mcbbgVtf0-6yGaH9-pKartllrwGaYPh3mm9kBWGWBDbtcklT_bUNTtYKqxLXay" alt="img" style="zoom:50%;" />

```

Comment:
I'm not sure if it could be displayed in China mainland. Any suggestion, @TomShawn ?

Reply:
Recommended to place images in the `media` directory of this repo and reference images in the docuemnts.
![image](https://user-images.githubusercontent.com/41534398/141290989-914186bf-b0a1-4ef1-baa6-232c22b54143.png)

Reply:
edit link to media